src/openvic-simulation/history/DiplomaticHistory.hpp

```diff
@@ -34,10 +35,9 @@ namespace OpenVic {
 
 		private:
 			Country const* PROPERTY(country);
-			Date PROPERTY_CUSTOM_PREFIX(joined, get_date);
-			std::optional<Date> PROPERTY_CUSTOM_PREFIX(exited, get_date);
+			Period period;
```

Comment:
```suggestion
			Period PROPERTY(period);
```
This is correctly non-const, because you need to update it when they exit the war 👍